### PR TITLE
Bump Deezer-python to 5.12, optimizations, quality of life changes. 

### PR DESCRIPTION
**Pull request by Un10ck3d**
This PR adds a few optimizations, bumps deezer-python to 5.12 and a few quality of life changes.
Optimizations include:
- Setting throttler to 50 requests per 5 seconds instead of 4 requests per 1 second. 
- Not checking if track is available that much and only when the track is getting parsed. 

Other changes:
- Setting name of the provider according to account name. This makes it easier to distinguish if multiple are added. 
- Bump deezer-python to 5.12.0 from 5.10.0
- Stop using eval to convert string to integer because its a security issue.

### CODE REVIEW

File: music_assistant/server/providers/deezer/__init__.py [hunk 1]
```diff
@@ -143,6 +143,7 @@ async def handle_setup(self) -> None:
 
         self.gw_client = GWClient(self.mass.http_session, self.config.get_value(CONF_ACCESS_TOKEN))
         await self.gw_client.setup()
+        self.config.name = f"Deezer - {self.client.user.name}"
```
Comment (marcelveldt) [inline music_assistant/server/providers/deezer/__init__.py:146]: this is actually not supposed to be changed this way, you are overriding the user's config.
instead override the name property:

```
@property
    def name(self) -> str:
        """Return (custom) friendly name for this provider instance."""
       if client := getattr(self, 'client', None):
         return f"Deezer - {client.user.name}"
       return super().name
```

File: music_assistant/server/providers/deezer/gw_client.py [hunk 2]
```diff
@@ -150,4 +150,54 @@ async def get_deezer_track_urls(self, track_id):
         if error := result_json["data"][0].get("errors"):
             raise DeezerGWError("Received an error from API", error)
 
+        print(await self.mark_track_played(track_id=track_id))
```
Comment (marcelveldt) [inline music_assistant/server/providers/deezer/gw_client.py:153]: please either drop the non working code or finish it

Comment (Un10ck3d) [reply music_assistant/server/providers/deezer/gw_client.py:153]: Yeah I was not planning to already do a PR but then the deezer-python version came out and and I wasn't home so I can't fix it yet.

**Comment by Un10ck3d** [comment]
With 50 requests per 5 seconds we can for example do 40 requests in 2 seconds and then only do 10 after that. But yes its true I don't use the throttler... I thought it would magically throttle stuff but yeah that dosnt make much sense i see.
But in general I made this pr way to early and just mashed thing together I think..

**Comment by marcelveldt** [comment]
@Un10ck3d I think at least we should make sure we fix the eventloop lockups asap as I still see a lot of sync requests blocking the loop. In fact I noticed it even hurting playback.

So, if you need some more time for the other stuff in the PR, maybe break it down into smaller chunks and PR now what is ready ?

**Comment by Un10ck3d** [comment]
So sorry I didn't see your message, I will do it tomorrow ASAP. 
About blocking sync code; yeah I have been trying to find the blocking come but without luck.

Maybe I should just port deezer-python to aiohttp 🤔

**Comment by Un10ck3d** [comment]
Closing this in favor of my new pr's

**Comment by jozefKruszynski** [comment]
> So sorry I didn't see your message, I will do it tomorrow ASAP. About blocking sync code; yeah I have been trying to find the blocking come but without luck.
> 
> Maybe I should just port deezer-python to aiohttp 🤔 Edit: Ok i did it now anyway: #681 #682 #683

I'm thinking about trying to do the same for tidalapi. 😬

**Pull request closed without merging**